### PR TITLE
add trim token filter docs #8449

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add trim token filter docs

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/8449

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _analyzers/token-filters/trim.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+title: Trim
+parent: Token filters
+nav_order: 430
+---
+
+# Trim token filter
+
+The `trim` token filter removes leading and trailing white space characters from tokens. 
+
+Many popular tokenizers, for example `standard`, `keyword`, and `whitespace` tokenizers, automatically strip any leading and trailing spaces during tokenization. When using these tokenizers, there is no need to configure an additional `trim` token filter.
```
Comment (natebower) [inline _analyzers/token-filters/trim.md:12]:
```suggestion
Many popular tokenizers, such as `standard`, `keyword`, and `whitespace` tokenizers, automatically strip leading and trailing white space characters during tokenization. When using these tokenizers, there is no need to configure an additional `trim` token filter. 
```

File: _analyzers/token-filters/index.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ Normalization | `arabic_normalization`: [ArabicNormalizer](https://lucene.apache
 `stop` | [StopFilter](https://lucene.apache.org/core/8_7_0/core/org/apache/lucene/analysis/StopFilter.html) | Removes stop words from a token stream.
 `synonym` | N/A | Supplies a synonym list for the analysis process. The synonym list is provided using a configuration file.
 `synonym_graph` | N/A | Supplies a synonym list, including multiword synonyms, for the analysis process.
-`trim` | [TrimFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TrimFilter.html) | Trims leading and trailing white space from each token in a stream. 
+[`trim`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/trim/) | [TrimFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TrimFilter.html) | Trims leading and trailing white space from each token in a stream.
```
Comment (natebower) [inline _analyzers/token-filters/index.md:60]:
```suggestion
[`trim`]({{site.url}}{{site.baseurl}}/analyzers/token-filters/trim/) | [TrimFilter](https://lucene.apache.org/core/9_10_0/analysis/common/org/apache/lucene/analysis/miscellaneous/TrimFilter.html) | Trims leading and trailing white space characters from each token in a stream. 
```

File: _analyzers/token-filters/trim.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+The `trim` token filter removes leading and trailing white space characters from tokens. 
+
+Many popular tokenizers, for example `standard`, `keyword`, and `whitespace` tokenizers, automatically strip any leading and trailing spaces during tokenization. When using these tokenizers, there is no need to configure an additional `trim` token filter. 
+{: .note}
+
+
+## Example
+
+The following example request creates a new index named `my_pattern_trim_index` and configures an analyzer with a `trim` filter and a `pattern` tokenizer, which does not remove leading and trailing white space:
```
Comment (natebower) [inline _analyzers/token-filters/trim.md:18]:
```suggestion
The following example request creates a new index named `my_pattern_trim_index` and configures an analyzer with a `trim` filter and a `pattern` tokenizer, which does not remove leading and trailing white space characters:
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws A few changes. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**